lhotse/lazy.py

```diff
@@ -363,11 +363,17 @@ def __init__(self, iterator: Iterable, predicate: Callable[[Any], bool]) -> None
     def __iter__(self):
         return filter(self.predicate, self.iterator)
 
-    # note: no __len__
-
     def __add__(self, other) -> "LazyIteratorChain":
         return LazyIteratorChain(self, other)
 
+    def __len__(self) -> int:
+        raise NotImplementedError(
```

Comment:
Can you check other lazy iterator classes and see if they can also use this error message? At least one that comes to my mind is LazyFlatten.

Reply:
Added to `LazyFlattener`. All other lazy iterator classes already have `__len__` defined.